### PR TITLE
Update Doctrine ORM

### PR DESCRIPTION
**Pull request by Sperling-0**
- Resolve the errors

Ticket: https://phabricator.wikimedia.org/T356744

### CODE REVIEW

**Comment by moiikana** [comment]
once this is merged, create a new release

**Review by moiikana — APPROVED** [review]

File: src/DataAccess/PaymentMigration/InsertingPaymentHandler.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ private function flush(): void {
 		$this->startTransactionIfNeeded();
 		$stmt = $this->entityManager->getConnection()->prepare( "UPDATE spenden SET payment_id=? WHERE id=?" );
 		foreach ( $this->paymentIdCollection as $donationId => $paymentId ) {
-			$stmt->executeQuery( [ $paymentId, $donationId ] );
+			$stmt->executeQuery();
```
Comment (Abban) [inline src/DataAccess/PaymentMigration/InsertingPaymentHandler.php:48]: You need to bind the 2 values here or they won't be inserted into the query

File: src/DataAccess/PaymentMigration/PaymentIdFinder.php [hunk 2]
```diff
@@ -24,7 +24,7 @@ public function findPaymentIdForDonation( int $donationId ): int {
 	}
 
 	private function getPaymentIdFromDb( int $donationId ): ?int {
-		$paymentId = $this->getPaymentIdStatement->executeQuery( [ $donationId ] )->fetchOne();
+		$paymentId = $this->getPaymentIdStatement->executeQuery()->fetchOne();
```
Comment (Abban) [inline src/DataAccess/PaymentMigration/PaymentIdFinder.php:27]: The donation ID needs to be insterted into the query here.

**Review by Abban — CHANGES_REQUESTED** [review]

File: src/DataAccess/PaymentMigration/InsertingPaymentHandler.php (hunk 1 above)
Comment (moiikana) [reply src/DataAccess/PaymentMigration/InsertingPaymentHandler.php:48]: the $donationId variable also wasn't used anywhere after that change.. I would have hoped that some code checker tooling would have prevented that :(

Comment (Sperling-0) [reply src/DataAccess/PaymentMigration/InsertingPaymentHandler.php]: Is my fix for this correct?

Comment (Abban) [reply src/DataAccess/PaymentMigration/InsertingPaymentHandler.php]: I think it needs to be:

```php
$statement->bindValue( 1, $paymentId );
$statement->bindValue( 2, $membershipId );
```

File: src/DataAccess/PaymentMigration/PaymentIdFinder.php (hunk 2 above)
Comment (Abban) [reply src/DataAccess/PaymentMigration/PaymentIdFinder.php:28]: This should be:

```php
$this->getPaymentIdStatement->bindValue( 1, $donationId );
```

**Review by Abban — APPROVED** [review]

**Pull request merged**